iFixit/react-commerce
# Product page SEO fields

closes #824 

This PR adds the `title`, `description` and `canonical` meta tags described in the issue.

This PR however also focused on including `og:image` tags to the product page.

We decided to add multiple `og:image`s since some sharing previews (like the facebook one) allow to view a gallery of images. 

Regarding the order in which images are written onto the page it was given priority to generic images over variant specific ones. According to [the protocol](https://ogp.me/#array) this means putting the generic ones first in the markup.

The reason of this is that, according to the spec, the content will be indexed according to the canonical, so probably most of the share preview implementations will include the image associated to the canonical independently of the fact that the link with parameters represents a specific variant url.
So a generic image, if any, would be more accurate to represent the content.
> og:url - The canonical URL of your object that will be used as its permanent ID in the graph, e.g., "https://www.imdb.com/title/tt0117500/".

At first we also included many optional parameters like `og:image:width`, `og:image:height`, `og:image:alt` etc but it came out that the Next Head component is not taking particular care of preserving the tag order and this could have been problematic. So we reverted the code to include the `og:image` tags only.

I've attached a video that shows how tag order is modified by client side interactions: updated tags are concatenated to the bottom of the list, separating each one of them from their siblings

https://user-images.githubusercontent.com/7805759/196474910-d0989f0f-c277-4ef1-b048-c569e601a87e.mov

### QA

1. Visit Vercel preview
2. Check that all the described SEO tags are present on the page
